app/controllers/SignUpController.scala

```diff
@@ -67,83 +74,87 @@ class SignUpController @Inject() (
-                    WebpageActivityTable.save(WebpageActivity(0, user.userId.toString, ipAddress, "SignIn", timestamp))
-
-                    env.eventBus.publish(SignUpEvent(user, request, request2lang))
-                    env.eventBus.publish(LoginEvent(user, request, request2lang))
-                    authenticator
-                  }
-                  request.authenticator match {
-                    case Some(oldAuthenticator) =>
-                      // If someone was already authenticated (i.e., they were signed into an anon user account), Play
-                      // doesn't let us sign one account out and the other back in in one response header. So we start
-                      // by redirecting to the "/finishSignUp" endpoint, discarding the old authenticator info and
-                      // sending the new authenticator info in a temp element in the session cookie. The "/finishSignUp"
-                      // endpoint will then move authenticator we put in "temp-authenticator" over to "authenticator"
-                      // where it belongs, finally completing the sign up.
-                      val redirectURL: String = nextUrl match {
-                        case Some(u) => "/finishSignUp?url=" + u
-                        case None => "/finishSignUp"
-                      }
-                      val result = newAuthenticator.map { newAuth =>
-                        // When we encrypt/serialize, we're doing the work of init, serialize, and embed. Found here:
-                        // https://github.com/mohiva/play-silhouette/blob/2.0.x/silhouette/app/com/mohiva/play/silhouette/impl/authenticators/SessionAuthenticator.scala
-                        val authSerialized: String = Crypto.encryptAES(Json.toJson(newAuth).toString())
-                        Redirect(redirectURL).withSession("temp-authenticator" -> authSerialized)
-                      }
-                      oldAuthenticator.discard(result)
-
-                    case None =>
-                      // If no account was signed in before, we can skip the "/finishSignUp" endpoint step. Instead, we
-                      // embed the authenticator into the session cookie in the normal way and either forward to the
-                      // new URL or simply send the newly authenticated user object.
-                      val result = nextUrl match {
-                        case Some(u) => Future.successful(Redirect(u))
-                        case None => Future.successful(Ok(Json.toJson(user)))
-                      }
-                      for {
-                        newA <- newAuthenticator
-                        value <- env.authenticatorService.init(newA)
-                        resultWithAuth <- env.authenticatorService.embed(value, result)
-                      } yield resultWithAuth
+            if (containsInvalidCharacters(data.username)) {
```

Comment:
Do you think that we should be checking if the username has invalid characters before checking if it already exists? I feel like we should be checking for invalid characters first unless you can think of a reason not to?